torch/optim/optimizer.py

```diff
@@ -491,8 +589,83 @@ def _process_value_according_to_param_policy(
             else:
                 return value.to(device=param.device)
 
+
+    def register_load_state_dict_pre_hook(
+        self,
+        hook: Callable[["Optimizer", StateDict], None],
+        prepend: bool = False,
+    ) -> RemovableHandle:
+        r"""Register a load_state_dict pre-hook which will be called before
+        :meth:`~torch.optim.Optimizer.load_state_dict` is called. It should have the
+        following signature::
+
+            hook(optimizer, state_dict) -> None
```

Comment:
Following the convention in the load_state_dict hooks in the module...those return None, but maybe we should diverge since being able to return it optionally strictly gives more options which seem better.

Reply:
Sounds weird not to allow returning it.

Reply:
Yep. Updating the initial issue with this point.